File: Ex_P1/app/src/main/java/com/example/ex_p1/MainActivity.java
package com.example.ex_p1;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Spinner;
import android.widget.TextView;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener {

    private ImageView ivMovie;
    private Spinner spSched;
    private String text;
    private String movie;
    private Intent change;
    private AlertDialog.Builder alert;
    private TextView tvTitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button bnChange = findViewById(R.id.bnChange);

        tvTitle = findViewById(R.id.tvTitle);
        spSched = findViewById(R.id.spSched);
        ivMovie = findViewById(R.id.ivMovie;

        //Cambiar color desde codigo
        //tvChannel.setBackgroundColor(ContextCompat.getColor(this, R.color.teal_700));

        spSched.setOnItemSelectedListener(this);
        bnChange.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                text = .getText().toString();
                if (text.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Escribe en la parte de arriba", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(MainActivity.this, "El valor era " + text, Toast.LENGTH_SHORT).show();
                    tvChannel.setText(text);
                }
            }
        });

        ivMovie.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                intent();
            }
        });
    }

    @Override
    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {

        movie = adapterView.getItemAtPosition(i).toString();
        Snackbar.make(adapterView, "Programa Seleccionado: " + movie, Snackbar.LENGTH_LONG).show();

        if (i == 0) {
            ivMovie.setImageResource(R.drawable.unnamed);
        } else {
            ivMovie.setImageResource(R.drawable.diasdetruenolarge);
        }
    }

    @Override
    public void onNothingSelected(AdapterView<?> adapterView) {

    }
    private void intent() {
        final CharSequence[] options = {"Ok", "Cancelar"};
        alert = new AlertDialog.Builder(MainActivity.this);
        alert.setTitle("Estas seguro de comprar los boletos?");
        alert.setItems(options, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                if (options[i].equals("Ok")) {
                    change = new Intent(MainActivity.this, SecondActivity.class);
                    text = tvTitle.getText().toString();
                    change.putExtra("Programa", movie);
                    change.putExtra("Canal", text);
                    startActivity(change);
                } else {
                    dialogInterface.dismiss();
                }

            }
        });
        alert.show();

    }
    }
}

File: Linear_Layout/app/src/main/java/edu/iest/linear_layout/MainActivity.kt
package edu.iest.linear_layout

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private var bnSuma : Button? = null
    private var bnResta : Button? = null
    private var bnMulti : Button? = null
    private var bnDiv : Button? = null
    private var etFirstNum : EditText? = null
    private var etSecondNum : EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        startViews()
        eventAssing()
    }

    private fun startViews() {
        bnSuma = findViewById(R.id.bnSuma)
        bnResta = findViewById(R.id.bnResta)
        bnMulti = findViewById(R.id.bnMult)
        bnDiv = findViewById(R.id.bnDiv)
        etFirstNum = findViewById(R.id.etFirstNum)
        etSecondNum = findViewById(R.id.etSecondNum)
    }

    private fun eventAssing() {
        bnSuma?.setOnClickListener(this)
        bnResta?.setOnClickListener(this)
        bnMulti?.setOnClickListener(this)
        bnDiv?.setOnClickListener(this)
    }

    override fun onClick(p0: View?) {
        //Entrara los clicks de las vistas
        val firstNum = etFirstNum?.text.toString().toFloatOrNull()
        val secondNum = etSecondNum?.text.toString().toFloatOrNull()
        val arit = Aritmetica()
        when(p0?.id) {
            R.id.bnSuma -> {
                if (firstNum != null && secondNum != null) {
                    val plus = arit.plus(firstNum, secondNum)
                    val msg = Mensajes("La suma es $plus")
                    msg.showToast(this@MainActivity)
                } else {
                    Toast.makeText(this, "No asignaste ningun número", Toast.LENGTH_LONG).show()

                }
            }
            R.id.bnResta -> {
                if (firstNum != null && secondNum != null) {
                    val minus = arit.minus(firstNum, secondNum)
                    val msg = Mensajes("La resta es $minus")
                    msg.showToast(this@MainActivity)
                } else {
                    Snackbar.make(p0, "No asignaste ninun número", Snackbar.LENGTH_LONG).show()
                }
            }
            R.id.bnMult -> {
                if (firstNum != null && secondNum != null) {
                    val minus = arit.mult(firstNum, secondNum)
                    val msg = Mensajes("La resta es $minus")
                    msg.showToast(this@MainActivity)
                } else {
                    Snackbar.make(p0, "No asignaste ninun número", Snackbar.LENGTH_LONG).show()
                }
            }
            R.id.bnDiv -> {
                if (firstNum != null && secondNum != null) {
                    val minus = arit.div(firstNum, secondNum)
                    val msg = Mensajes("La resta es $minus")
                    msg.showToast(this@MainActivity)
                } else {
                    Snackbar.make(p0, "No asignaste ninun número", Snackbar.LENGTH_LONG).show()
                }
            }
        }
    }
}